File: 6_files/4.py
import random
import string
import os

def generate():
    if "f1.txt" in os.listdir("."):
        os.remove("f1.txt")

    f1 = open("f1.txt", "a")
    for _ in range(10):
        f1.write("".join(random.choice(string.digits) for _ in range(2)) + "\n")

    f1.close()

def counter(num):
    count = 0
    num = abs(num)
    for i in range(1,num):
        if num % i == 0:
            count += 1

    return count


def lookup():
    f1 = open("f1.txt", "r")
    
    out = []
    for ln in f1.read().splitlines():
        count = counter(int(ln))
        if count == 3:    
            out.append(ln)

        print(f"{ln} - {count} - {count==3}")

    f1.close()
    open("f2.txt", "w").write("\n".join(out))

generate()
lookup()

File: c/10/4.c
//Дан файл f содержащий целые числа. Записать в файл f1 компоненты
//исходного файла, имеющие ровно 3 делителя.

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void generate() {
    FILE *f1 = fopen("f1.txt", "w");
    if (f1 == NULL) {
        perror("Error opening file");
        return;
    }

    for (int i = 0; i < 10; i++) {
        for (int j = 0; j < 2; j++) {
            fputc('0' + rand() % 10, f1); // Random digit
        }
        fputc('\n', f1);
    }
    fclose(f1);
}

int counter(int num) {
    int count = 0;
    num = abs(num);
    for (int i = 1; i <= num; i++) {
        if (num % i == 0) {
            count++;
        }
    }
    return count;
}

void lookup() {
    FILE *f1 = fopen("f1.txt", "r");
    FILE *f2 = fopen("f2.txt", "w");

    char line[100];
    while (fgets(line, sizeof(line), f1)) {
        int num = atoi(line);
        int count = counter(num);
        if (count == 3) {
            fprintf(f2, "%s", line);
        }

        line[strcspn(line, "\n")] = 0;
        printf("%s - %d - %s\n", line, count, count == 3 ? "true" : "false");
    }

    fclose(f1);
    fclose(f2);
}

void main() {
    generate();
    lookup();
}